src/sqlancer/postgres/PostgresProvider.java

```diff
@@ -218,24 +220,51 @@ protected TestOracle getTestOracle(PostgresGlobalState globalState) throws SQLEx
 
     @Override
     public Connection createDatabase(PostgresGlobalState globalState) throws SQLException {
-        String url = "jdbc:postgresql://localhost:5432/test";
+        String username = globalState.getOptions().getUserName();
+        String password = globalState.getOptions().getPassword();
+        String entryPath = "test";
+        String entryURL = globalState.getDmbsSpecificOptions().connectionURL;
+        // trim URL to exclude "jdbc:"
+        if (entryURL.startsWith("jdbc:")) {
+            entryURL = entryURL.substring(5);
+        }
+        try {
+            URI uri = new URI(entryURL);
+            String userInfoURI = uri.getUserInfo();
+            String pathURI = uri.getPath();
+            if (userInfoURI != null) {
+                // username and password specified in URL take precedence
+                String[] userInfo = userInfoURI.split(":", 2);
+                username = userInfo[0];
+                password = userInfo[1];
+                int userInfoIndex = entryURL.indexOf(userInfoURI);
+                entryURL = entryURL.substring(0, userInfoIndex)
+                        + entryURL.substring(userInfoIndex + userInfoURI.length() + 1);
+            }
+            if (pathURI != null) {
+                entryPath = pathURI;
+            }
+        } catch (URISyntaxException e) {
+            throw new AssertionError(e);
+        }
         String databaseName = globalState.getDatabaseName();
-        Connection con = DriverManager.getConnection(url, globalState.getOptions().getUserName(),
-                globalState.getOptions().getPassword());
-        globalState.getState().logStatement("\\c test;");
+        Connection con = DriverManager.getConnection("jdbc:" + entryURL, username, password);
+        globalState.getState().logStatement(entryURL);
```

Comment:
What is the entry URL here? If we log it without appending a `-- ` first, is the log still executable as a SQL script?

Reply:
This is the connection string. Does logging involve executing SQL statements? I thought they were for printing, more clarification on that would be great in case I missed something!

Reply:
Things written to the log should typically either be SQL statements or comments (prefixed by `--`), so that the log file can basically be directly used as an input to a DBMS. I should update the documentation about this.